File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    completion_map = dict()
    for name in completion:
        if name in completion_map:
            completion_map[name] = completion_map[name] + 1
        else:
            completion_map[name] = 1
    
    for name in participant:
        if name not in completion_map or completion_map[name] == 0:
            return name
        else:
            completion_map[name] = completion_map[name] - 1
            
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Arrays;

class Solution {
    public String solution(String[] participant, String[] completion) {  
    	Arrays.sort(participant);
    	Arrays.sort(completion);
    	String answer = participant[completion.length];  
    	for(int i=0; i<completion.length; i++) {
    			if(!participant[i].equals(completion[i])) {
    				answer = participant[i];
                    break;
    			}
    	} 	
        return answer;
    }
}